### PR TITLE
chore: add duplicate-external-PR GH action

### PR DESCRIPTION
**Pull request by joaoviana**
### Description:

Adds `Duplicate External PR` by commenting on PR with `/duplicate-external-pr`

Before doing so, it checks if user belongs to `lightdash/engineering` team

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: .github/workflows/duplicate-external-pr.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.issue.pull_request.head.ref }}
+    - name: Duplicate external PR
+      run: |
+        git config user.name "github-actions[bot]"
+        git config user.email "github-actions[bot]@users.noreply.github.com"
+        git checkout -b duplicate-${{ github.event.issue.number }} ${{ github.event.issue.pull_request.head.ref }}
+        git push origin duplicate-${{ github.event.issue.number }}
+        gh pr create --base ${{ github.event.repository.default_branch }} --head duplicate-${{ github.event.issue.number }} --title "[Duplicate] ${{ github.event.issue.title }}" --body "This is a duplicated PR for running E2E tests." --assignee ${{ github.event.comment.user.login }}
```
[the remark this answers is not in the document]
Comment (joaoviana) [reply .github/workflows/duplicate-external-pr.yml]: done here: https://github.com/lightdash/lightdash/pull/8986/commits/8caa545c33bb1c908cc3f0e4148c9ab6653be664

File: .github/workflows/duplicate-external-pr.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    issue_comment:
+        types: [created]
+jobs:
+    duplicate-and-test:
+        if: github.event.issue.pull_request && contains(github.event.comment.body, '/duplicate-external-pr')
+        runs-on: ubuntu-latest
+        steps:
+            - name: Check user for team affiliation
+              id: check-user-for-team-affiliation
+              uses: tspascoal/get-user-teams-membership@v3
```
Comment (joaoviana) [inline .github/workflows/duplicate-external-pr.yml:12]: https://github.com/tspascoal/get-user-teams-membership?tab=readme-ov-file

File: .github/workflows/duplicate-external-pr.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+              env:
+                  ISSUE_NUMBER: ${{ github.event.issue.number }}
+                  PR_TITLE: ${{ github.event.issue.title }}
+                  REF: ${{ github.event.issue.pull_request.head.ref }}
+                  DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
+                  ASSIGNEE: ${{ github.event.comment.user.login }}
+              run: |
+                  git config user.name "github-actions[bot]"
+                  git config user.email "github-actions[bot]@users.noreply.github.com"
+                  git checkout -b duplicate-${{ env.ISSUE_NUMBER }} ${{ env.REF }}
```
Comment (ZeRego) [inline .github/workflows/duplicate-external-pr.yml:37]: Would be awesome to support the case where the branch already exists.
For the cases when we ask for changes, we could run the command to update the branch/PR.

**Review by ZeRego — APPROVED** [review]

File: .github/workflows/duplicate-external-pr.yml (hunk 3 above)
Comment (joaoviana) [reply .github/workflows/duplicate-external-pr.yml]: Great! I've added this check as well.

Checks if branch exists already - if so, update the branch


https://github.com/lightdash/lightdash/pull/8986/commits/0e55f10760709a5467c17b8a25d1e0a7ee686e8a

**Pull request merged**